src/profile/helpers/editData.ts

```diff
@@ -334,8 +341,54 @@ export function updateItems(
   return dataHasUpdated ? newList : null;
 }
 
+// this is a temporary function to assure primaryItem is always at index 0
+// remove this when backend orders them
```

Comment:
Order by id is fine. UI will anyway put the primary item as first. Items should be ordered by some attribute/value -pair that is returned in GraphQL, so UI can order them too after primary has changed.  This way UI can order them immediately without waiting for backend.

id is a string. Is it alphabetically sortable and newest is last in A-Z ordering? It would be confusing for the user if newest would jump to other position than last.

Reply:
> Order by id is fine. UI will anyway put the primary item as first. Items should be ordered by some attribute/value -pair that is returned in GraphQL, so UI can order them too after primary has changed. This way UI can order them immediately without waiting for backend.
> 
> id is a string. Is it alphabetically sortable and newest is last in A-Z ordering? It would be confusing for the user if newest would jump to other position than last.

In the database the `id` is numeric and increases for new items so that's a natural stable key for sorting. The `id` returned via GraphQL is based on the database `id`: it's prefixed with some static string and then base64 encoded. I guess mostly sorting those alphabetically leads to the same result, but there could be cases where that doesn't quite work. But it's not guaranteed that the GraphQL `id` calculation algorithm will remain as described here. It's an implementation detail and shouldn't be relied on.

But does UI need to do full sorting? It has the list it gets from server. It shows the primary item as first but otherwise the list order is the same. When primary item is changed, the used-to-be-primary item returns to its original place in the UI and the new primary gets to be shown first. That should match the order received from server, right?

Reply:
Sounds good, when sorted with id! UI has to either save the original order of ids or keep the ordering and re-order primary only when data is rendered. The latter sounds better at the moment.

I'll add this here right now., but backend has to be changed too. Before backend sorts by "id", there is some glitches in UI when items change place. But that won't show unless user has 3 or more items.

Reply:
Yes, putting primary first at the "very last moment" in the UI sounds sensible to me too. So it's just for visualization. Everywhere else in the UI project code the order stays unchanged.